pkg/storage/unified/resource/server.go

```diff
@@ -294,16 +297,16 @@ func (s *server) Init(ctx context.Context) error {
 			}
 		}
 
-		// Start watching for changes
-		if s.initErr == nil {
-			s.initErr = s.initWatcher()
-		}
-
 		// initialize the search index
 		if s.initErr == nil && s.search != nil {
 			s.initErr = s.search.init(ctx)
 		}
 
+		// Start watching for changes
```

Comment:
I think we want to finish building the index before we start receiving watch events